src/main/java/org/cloudfoundry/promregator/auth/PromregatorHttpClientFactory.java

```diff
@@ -0,0 +1,46 @@
+package org.cloudfoundry.promregator.auth;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+import javax.net.ssl.SSLContext;
+
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sap.cloud.security.client.HttpClientException;
+import com.sap.cloud.security.config.ClientIdentity;
+import com.sap.cloud.security.mtls.SSLContextFactory;
+
+public class PromregatorHttpClientFactory implements com.sap.cloud.security.client.HttpClientFactory {
+
+	private static final Logger log = LoggerFactory.getLogger(PromregatorHttpClientFactory.class);
+
+	@Override
+	public CloseableHttpClient createClient(ClientIdentity clientIdentity) throws HttpClientException {
```

Comment:
I think we should mention the source of this coding here in a comment.

Reply:
done